### ISSUE TITLE
split for nas

### ISSUE THREAD

**Issue by HeathHose** [yangsenius/PoseNFS#2]
Thanks for your awesome work.
I notice that some code is used to keep the same iterations in train and search

```assert train_iters==valid_iters```
I can't understand it. Is it a convention in nas?

**Comment by yangsenius** [comment]
Thanks for your attention. This setting comes from the update rule of DARTS. We use this code to make sure that the number of update steps (train iters) for the weights is equal with the number of update steps (valid iters) for the architecture parameters, no matter of what proportion between train set and valid set.

**Comment by yangsenius** [comment]
Thanks for your attention. This setting comes from the update rule of DARTS. We need to make sure that the number of steps (train iters) for updating the weights is equal with the number of steps (valid iters) for updating the architecture parameters, no matter what proportion between the sizes of training set and valid set.

**Comment by yangsenius** [comment]
Thanks for your attention. This setting comes from the update rule of DARTS. We need to make sure that the number of steps (train iters) for updating the weights is equal with the number of steps (valid iters) for updating the architecture parameters, no matter what proportion between the sizes of training set and valid set. Note that  `valid iters` here means the length of `arch_queue`, not `valid_queue`, it maybe a little  confusing.

**Comment by yangsenius** [comment]
Thanks for your attention. This setting comes from the update rule of DARTS. We need to make sure that the number of steps (train iters) for updating the weights is equal with the number of steps (valid iters) for updating the architecture parameters, no matter what proportion between the sizes of training set and valid set. Note that valid iters here means the length of arch_queue, not valid_queue, it may be a little confusing.

**Comment by HeathHose** [comment]
@yangsenius  Thanks for your reply. But there is no description but ```To carry out architecture search, we hold out half of the CIFAR-10 training data as the validation set.``` in paper
Maybe, this is just a coincidence

**Comment by yangsenius** [comment]
DARTS `hold out half of the CIFAR-10 training data as the validation set`  and set the same batchsize, then
`len(train set)==len(valid set)`==>` len(train set)/batchsize==len(valid set)/batchsize` ==>`train_iters==valid_iters`.
